### PR TITLE
[move-2024] Method syntax for `move-stdlib`

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

Stacked on top of #16634, does what it says on the tin (and it makes the code all shiny!)

Public use funs:

```
ascii.move
10:    public use fun std::string::from_ascii as String.to_string;

vector.move
10:    public use fun std::string::utf8 as vector.to_string;
14:    public use fun std::string::try_utf8 as vector.try_to_string;
17:    public use fun std::ascii::string as vector.to_ascii_string;
21:    public use fun std::ascii::try_string as vector.try_to_ascii_string;
```

## Test Plan 

All the tests still work as expected

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by tzakian — APPROVED** [review]
A.M.A.Z.I.N.G. 🚀 

It's looking so pretty with the new syntax! I mean how can you not shed a little tear of joy when you see things like this

```diff
-  vector::length(as_bytes(string))
+  string.as_bytes().length()
````

File: crates/sui-framework/packages/move-stdlib/sources/vector.move [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
+    /// Allows calling `.try_to_string()` on a vector of `u8` to get a utf8 `String`.
+    /// This will return `None` if the vector is not valid utf8.
+    public use fun std::string::try_utf8 as vector.try_to_string;
+
+    /// Allows calling `.to_ascii_string()` on a vector of `u8` to get an `ascii::String`.
+    public use fun std::ascii::string as vector.to_ascii_string;
+
+    /// Allows calling `.try_to_ascii_string()` on a vector of `u8` to get an
+    /// `ascii::String`. This will return `None` if the vector is not valid ascii.
+    public use fun std::ascii::try_string as vector.try_to_ascii_string;
```
Comment (tnowacki) [inline crates/sui-framework/packages/move-stdlib/sources/vector.move:21]: Y'alls thoughts shortening to `to_ascii`?

**Review by tnowacki — COMMENTED** [review]
overall LGTM, just one comment about naming

File: crates/sui-framework/packages/move-stdlib/tests/bcs_tests.move [hunk 2]
```diff
@@ -101,7 +101,8 @@ module std::bcs_tests {
     }
 
     #[test]
-    #[expected_failure(abort_code = 453, location = std::bcs)]
+    #[expected_failure]
```
Comment (tnowacki) [inline crates/sui-framework/packages/move-stdlib/tests/bcs_tests.move:104]: Why the dropping of the location and code?

Comment (cgswords) [reply crates/sui-framework/packages/move-stdlib/tests/bcs_tests.move:104]: The error is now that this hits the VM object depth, not the `bcs` decoding error that was previously being thrown.

**Review by tnowacki — APPROVED** [review]

File: crates/sui-framework/packages/move-stdlib/sources/vector.move (hunk 1 above)
Comment (cgswords) [reply crates/sui-framework/packages/move-stdlib/sources/vector.move:21]: I like `to_ascii_string` -- `to_ascii` might imply it hands back a vector.

**Pull request merged**